### PR TITLE
First push of Readme

### PR DESCRIPTION
**Pull request by chipsnyder**
Initial push of ReadMe discussed on during internal meeting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,684 @@
...
+**Not:**
+```objc
+[view setBackgroundColor:[UIColor orangeColor]];
+UIApplication.sharedApplication.delegate;
+```
+
+**Note:** Method notation is preferred when accessing an overridden getter or setter. This is in order to provide clarity that the value is being modified in some way before being set or returned. 
+
+When accessing a property on self dot notation should be used. Expect in the case of an overridden getter or setter.
+
```
Comment (MichaelCampbell) [inline README.md]: These two lines seem to be saying the same thing, maybe we can get rid of the line starting with "When"
>**Note:** Method notation is preferred when accessing an overridden getter or setter. This is in order to provide clarity that the value is being modified in some way before being set or returned. 
>
>When accessing a property on self dot notation should be used. Expect in the case of an overridden getter or setter.

Comment (chipsnyder) [inline README.md]: Reworded the line. I wanted to draw attention to the situation of using a property on `self` let me know if this is more clear or should be reworked again

File: README.md [hunk 2]
```diff
@@ -1,2 +1,684 @@
...
+**Not:**
+
+```objc
+static const NSString *brandNameHollister = @"Hollister";
+```
+
+Properties and local variables should be camel-case with the leading word being lowercase.
+
+Instance variables should be camel-case with the leading word being lowercase, and should be prefixed with an underscore. This is consistent with instance variables synthesized automatically by LLVM. **If LLVM can synthesize the variable automatically, then let it.**
```
Comment (hkellaway) [inline README.md]: i was a bit confused why this was in here as the guide says we prefer properties to instance variables

File: README.md [hunk 3]
```diff
@@ -1,2 +1,684 @@
...
+
+**For example:**
+```objc
+self.backgroundColor = [UIColor orangeColor];
+self.delegate;
+```
+
+**Not:**
+```objc
+[self setBackgroundColor:[UIColor orangeColor]];
```
Comment (MichaelCampbell) [inline README.md]: Maybe we should use another example here. I'm not sure how often we'd be overriding the setter for a UIColor :-)

File: README.md (hunk 2 above)
Comment (chipsnyder) [inline README.md]: I chose to leave this only because we do allow instance variables for initializers

File: README.md (hunk 3 above)
Comment (chipsnyder) [inline README.md]: That's why it's in the *Not* section

File: README.md [hunk 4]
```diff
@@ -1,2 +1,686 @@
...
+if (!someObject) { }
+if (someObject == nil) { }
+```
+
+**For a `BOOL` value:**
+
+```objc
+if (isAwesome) { }
+if (!someNumber.boolValue) { }
+if (someNumber.boolValue == NO) { }
```
Comment (hkellaway) [inline README.md]: if we're disallowing `isAwesome == YES`, should we not also disallow `isAwesome == NO`?

File: README.md (hunk 3 above)
Comment (MichaelCampbell) [inline README.md]: ooooooooh, gotcha.

File: README.md (hunk 4 above)
Comment (chipsnyder) [inline README.md]: Really we're disallowing a comparison to `YES` but I'm for disallowing `isAwesome == NO` as well. @MichaelCampbell @zgeorge Do you two agree as well?

Comment (MichaelCampbell) [inline README.md]: i totally agree

**Comment by chipsnyder** [comment]
@MichaelCampbell @hkellaway @zgeorge 

I think all that's outstanding is the question here:
https://github.com/AbercrombieAndFitch/Objective-C-Style-Guide/pull/1#discussion_r31238742

Since instance variables are allowed in `init` I left this in place. What are other opinions? Should we remove this section?

File: README.md (hunk 2 above)
Comment (MichaelCampbell) [inline README.md]: I think it might be worthwhile to leave this text, as long as we also list the use cases for the instance variables.

Comment (chipsnyder) [inline README.md]: Yeah we cover that in a section before this with:
https://github.com/AbercrombieAndFitch/Objective-C-Style-Guide/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R241

Comment (MichaelCampbell) [inline README.md]: Good point, i guess i glossed over that part. This should be good to go then.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,686 @@
-# Objective-C-Style-Guide
-Based off the NYTimes style guide (https://github.com/NYTimes/objective-c-style-guide). This style guide was morphed internally for A&amp;F.
+# Abercrombie & Fitch: Objective-C Style Guide
+
+This style guide outlines the coding conventions of the iOS teams at A&F. This is base on the [NYTimes Objective-C Style Guide](https://github.com/NYTimes/objective-c-style-guide)
```
Comment (MichaelCampbell) [inline README.md]: typo base -> based

Comment (chipsnyder) [inline README.md]: Thanks!

**Comment by MichaelCampbell** [comment]
:+1:

**Comment by hkellaway** [comment]
:+1:

**Pull request merged**